packages/SwingSet/src/vats/network/network.js

```diff
@@ -568,14 +637,21 @@ export function makeLoopbackProtocolHandler(E = defaultE) {
       }
       const [lport, lhandler] = listeners.get(remoteAddr);
       // console.log(`looking up onAccept in`, lhandler);
-      const rport = await E(lhandler).onAccept(
-        lport,
-        remoteAddr,
-        localAddr,
-        lhandler,
-      );
-      // console.log(`rport is`, rport);
-      return rport;
+      const remoteSuffix =
+        /** @type {Endpoint} */
+        (await E(lhandler)
+          .onInbound(lport, remoteAddr, localAddr, lhandler)
+          .catch(e => rethrowUnlessMissing(e)));
+
+      if (remoteSuffix) {
+        remoteAddr = `${remoteAddr}/${remoteSuffix}`;
+      }
+
+      const rchandler =
+        /** @type {ConnectionHandler} */
+        (await E(lhandler).onAccept(lport, remoteAddr, localAddr, lhandler));
```

Comment:
I’m surprised the linter didn’t strip the outer parentheses.

Reply:
It didn't because it's a legitimate syntax for jsdoc typecast.

Reply:
It didn't because it's a legitimate syntax for jsdoc typecast.

Reply:
Oh, the parentheses are affiliated with the comment and eslint knows about this? That’s intriguing.